quarkus/server/build.gradle.kts

```diff
@@ -91,7 +91,7 @@ distributions {
       from(project.layout.buildDirectory.dir("quarkus-app"))
       from("distribution/NOTICE")
       from("distribution/LICENSE")
-      exclude("lib/main/io.quarkus.quarkus-container-image*")
```

Comment:
FYI I tried to keep these jars excluded while still producing the Docker image during build, but my Gradle skills weren't enough 😆 – so I'm fine with this.

Reply:
That's not so easy with gradle for sure !

Reply:
Does it work if we uses runtimeOnly here? 
```
  implementation("io.quarkus:quarkus-container-image-docker")
```

Reply:
The dependencies are actually already there but excluded. So it's more buildOnly more than runtimeOnly 😄

Reply:
`runtimeOnly` wouldn't work. I tried the `buildOnly` approach btw, but it's too convoluted.

Reply:
With `runtimeOnly`, we need to customize the package task a bit like 
```
    from(configurations.runtimeOnly.get().filter {
        it.name.contains("quarkus-container-image-docker")
    }.map {
        zipTree(it)
    })
```

Reply:
That's not the problem: here we excluded the dependency because it was supposed to be a compile only dependency. But it's not. So we have to not exclude (whatever the scope is).

Reply:
The `quarkus-container-image-docker` is actually not needed at compile time. There is no compile time dependency on it. This is verified by my test by just removing it. It's needed and only needed at runtime per my test. With that, marking it as `runtimeOnly` is reasonable. The only problem here is that the packaging task doesn't pick runtimeOnly dependencies, that's why I propose to customize a bit to include it.

Reply:
```
./gradlew \
  :polaris-quarkus-server:assemble \
  :polaris-quarkus-server:quarkusAppPartsBuild --rerun \
  -Dquarkus.container-image.build=true
```
After remove the dependency(`//  implementation("io.quarkus:quarkus-container-image-docker")`
), this command works and built a local docker image successfully. Which docker image build will failed
```
apache/polaris                            0.11.0-beta-incubating-SNAPSHOT   7e1fc7ac2034   2 minutes ago   611MB
```

Reply:
I do not see any image being created. You can double-check by changing the image name:

```
./gradlew \
  :polaris-quarkus-server:assemble \
  :polaris-quarkus-server:quarkusAppPartsBuild --rerun \
  -Dquarkus.container-image.build=true -Dquarkus.container-image.name=foo
```

After that, I do not see any image named `foo` on my machine.

Reply:
Can you try this? It works for me.
```
  runtimeOnly("io.quarkus:quarkus-container-image-docker")
```

Reply:
`runtimeOnly` works, yes. If everybody thinks that's the best solution, I'm +1 on that.

Reply:
For now, I would keep "simple". I propose to do another PR to improve that. Thoughts ?